tools/authz_generate_rules_for_roles/main.go

```diff
@@ -0,0 +1,335 @@
+Steps to use:
+  - cd tools && go generate
+  - make lint-markdown-fix
+  - check diff for ./modules/140-user-authz/docs/README.md and ./modules/140-user-authz/docs/README_RU.md files
+*/
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	readVerbs      = []string{"get", "list", "watch"}
+	writeVerbs     = []string{"create", "delete", "deletecollection", "patch", "update"}
+	readWriteVerbs = append(readVerbs, writeVerbs...)
+
+	readVerbsString      = sliceToString(readVerbs)
+	writeVerbsString     = sliceToString(writeVerbs)
+	readWriteVerbsString = sliceToString(readWriteVerbs)
+
+	readAlias      = "read"
+	writeAlias     = "write"
+	readWriteAlias = "read-write"
+
+	clusterRoleModulePrefix = "user-authz:"
+
+	userString           = "user"
+	privilegedUserString = "privileged-user"
+	editorString         = "editor"
+	adminString          = "admin"
+	clusterEditorString  = "cluster-editor"
+	clusterAdminString   = "cluster-admin"
+
+	userClusterRole           = clusterRoleModulePrefix + userString
+	privilegedUserClusterRole = clusterRoleModulePrefix + privilegedUserString
+	editorClusterRole         = clusterRoleModulePrefix + editorString
+	adminClusterRole          = clusterRoleModulePrefix + adminString
+	clusterEditorClusterRole  = clusterRoleModulePrefix + clusterEditorString
+	clusterAdminClusterRole   = clusterRoleModulePrefix + clusterAdminString
+
+	orderedRoleNames = []string{
+		userClusterRole, privilegedUserClusterRole,
+		editorClusterRole, adminClusterRole,
+		clusterEditorClusterRole, clusterAdminClusterRole,
+	}
+
+	neededClusterRoleExcludes = map[string][]string{
+		userClusterRole:           {},
+		privilegedUserClusterRole: {userClusterRole},
+		editorClusterRole:         {userClusterRole, privilegedUserClusterRole},
+		adminClusterRole:          {userClusterRole, privilegedUserClusterRole, editorClusterRole},
+		clusterEditorClusterRole:  {userClusterRole, privilegedUserClusterRole, editorClusterRole},
+		clusterAdminClusterRole:   {userClusterRole, privilegedUserClusterRole, editorClusterRole, adminClusterRole, clusterEditorClusterRole},
+	}
+)
+
+func main() {
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	if err := os.Chdir("../"); err != nil {
+		log.Fatalln(err)
+	}
+
+	renderContents, err := renderTemplates()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	if err := os.Chdir(cwd); err != nil {
+		log.Fatalln(err)
+	}
+
+	clusterRolesMap := getClusterRoles(renderContents)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	crVerbResourceMap := make(map[string]map[string][]string)
+	for name, rules := range clusterRolesMap {
+		if _, f := neededClusterRoleExcludes[name]; !f {
+			continue
+		}
+		crVerbResourceMap[name] = processClusterRoleRules(rules)
+	}
+
+	builder := &strings.Builder{}
+	builder.WriteString("* read - `get`, `list`, `watch`\n")
```

Comment:
Instead of using a text builder, it is better to use `text/template` here in terms of readability.